tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala

```diff
@@ -273,8 +322,23 @@ private[scalajs] final class ScalaJSClassEmitter(
       val js.Function(params, body) = genConstructorFunForJSClass(tree)
       js.MethodDef(static = false, js.Ident("constructor"), params, body)
     } else {
+      val className = tree.name.name
       val fieldDefs = genFieldDefs(tree)
-      if (fieldDefs.isEmpty && outputMode == OutputMode.ECMAScript6) {
+
+      val (otherStatements, args) = if (usesJSConstructorOpt(className)) {
+        val (ctor :: Nil) = tree.memberMethods.map(_.tree).filter(
+            x => isConstructorName(x.name.name))
+        val js.Function(args, ctorBody) = desugarToConstructor(this,
+            tree.name.name, ctor.name.name, ctor.static,
+            ctor.args, ctor.body, js.Skip())
+        (ctorBody, args)
+      } else {
+        (js.Skip(), Nil)
+      }
+
+      if (fieldDefs.isEmpty &&
+        outputMode == OutputMode.ECMAScript6 &&
+        otherStatements.isInstanceOf[js.Skip]) {
```

Comment:
The condition of an `if` is not allowed to have 3 lines. Keep the `outputMode` test on the first line with `fieldDefs.isEmpty`.

Also: 4 spaces.